packages/controllers/src/plugins/PluginController.test.ts

```diff
@@ -0,0 +1,122 @@
+import fs from 'fs';
+import { ControllerMessenger } from '@metamask/controllers/dist/ControllerMessenger';
+import { WorkerController } from '../workers/WorkerController';
+import { PluginController } from './PluginController';
+
+const workerCode = fs.readFileSync(
+  require.resolve('@mm-snap/workers/dist/PluginWorker.js'),
+  'utf8',
+);
+
+describe('PluginController Controller', () => {
+  it('can create a worker and plugin controller', async () => {
+    const workerController = new WorkerController({
+      setupWorkerConnection: jest.fn(),
+      workerUrl: new URL(URL.createObjectURL(new Blob([workerCode]))),
+    });
+    const pluginController = new PluginController({
+      command: workerController.command.bind(workerController),
+      createPluginWorker: workerController.createPluginWorker.bind(
+        workerController,
+      ),
+      terminateAll: workerController.terminateAll.bind(workerController),
+      terminateWorkerOf: workerController.terminateWorkerOf.bind(
+        workerController,
+      ),
+      startPlugin: workerController.startPlugin.bind(workerController),
+      removeAllPermissionsFor: jest.fn(),
+      getPermissions: jest.fn(),
+      hasPermission: jest.fn(),
+      requestPermissions: jest.fn(),
+      closeAllConnections: jest.fn(),
+      messenger: new ControllerMessenger<any, any>().getRestricted({
+        name: 'PluginController',
+        allowedActions: [],
+        allowedEvents: [],
+      }),
+      state: {
+        pluginStates: {},
+        plugins: {},
+      },
+      metadata: {
+        pluginStates: {
+          persist: false,
+          anonymous: false,
+        },
+        plugins: {
+          persist: false,
+          anonymous: false,
+        },
+      },
+    });
+    expect(pluginController).toBeDefined();
+  });
+  it('can add a plugin and use its JSON-RPC api', async () => {
+    const workerController = new WorkerController({
+      setupWorkerConnection: jest.fn(),
+      workerUrl: new URL(URL.createObjectURL(new Blob([workerCode]))),
+    });
+    const pluginController = new PluginController({
+      command: workerController.command.bind(workerController),
+      createPluginWorker: workerController.createPluginWorker.bind(
+        workerController,
+      ),
+      terminateAll: workerController.terminateAll.bind(workerController),
+      terminateWorkerOf: workerController.terminateWorkerOf.bind(
+        workerController,
+      ),
+      startPlugin: workerController.startPlugin.bind(workerController),
+      removeAllPermissionsFor: jest.fn(),
+      getPermissions: jest.fn(),
+      hasPermission: jest.fn(),
+      requestPermissions: jest.fn(),
+      closeAllConnections: jest.fn(),
+      messenger: new ControllerMessenger<any, any>().getRestricted({
+        name: 'PluginController',
+        allowedActions: [],
+        allowedEvents: [],
```

Comment:
`allowedActions` and `allowedEvents` are required in typescript:

![image](https://user-images.githubusercontent.com/364566/119179663-0cb96e00-ba24-11eb-8370-db64481432f4.png)

Reply:
I think this was improved in v9.0.0 - they aren't supposed to be required